tools/ipapy2/ipapy2_MS2_annotation.xml

```diff
@@ -0,0 +1,121 @@
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_MS2_annotation_cli">
+@init_logger@
+
+df = pd.read_csv('${MS1_table}')
+dfMS2 = pd.read_csv('${MS2_table}')
+allAdds = pd.read_csv('${all_adducts}')
+DBMS2 = pd.read_csv('${DBMS2_table}')
+annotations = ipa.MSMSannotation(df, dfMS2, allAdds, DBMS2, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, mzdCS=$mzdCS, ppmCS=$ppmCS, CSunk=$CSunk, evfilt=$evfilt, ncores=$ncores)
+annotations.to_csv('${output}', index=False)
+</configfile>
+</configfiles>
+
+    <inputs>
+        <param label="MS1 table" name="MS1_table" type="data" format="csv" help="pandas dataframe containing the MS1 data." />
+        <param label="MS2 table" name="MS2_table" type="data" format="csv" help="pandas dataframe containing the MS2 data." />
+        <param label="all possible adducts table" name="all_adducts" type="data" format="csv" help="pandas dataframe containing the information on all the possible adducts given the database." />
+        <param label="DBMS2 table" name="DBMS2_table" type="data" format="csv" help="pandas dataframe containing the database containing the MS2 information." />
+        <param label="ppm" name="ppm" type="float" help="accuracy of the MS instrument used."/>
+        <param label="me" name="me" type="float" value="0.000548579909065" help="accurate mass of the electron. Default 5.48579909065e-04."/>
+        <param label="ratiosd" name="ratiosd" type="float" value="0.9" help="It represents the acceptable ratio between predicted intensity and observed intensity of isotopes. default 0.9."/>
+        <param label="ppmunk" name="ppmunk" type="float" value='$ppm' help="ppm associated to the 'unknown' annotation. If not provided equal to ppm.."/>
+        <param label="ratiounk" name="ratiounk" type="float" value="0.5" help="isotope ratio associated to the 'unknown' annotation. If not provided equal to 0.5."/>
+        <param label="ppmthr" name="ppmthr" type="float" value='$ppm*2' help="Maximum ppm possible for the annotations. Ff not provided equal to 2*ppm."/>
+        <param label="pRTNone" name="pRTNone" type="float" value="0.8" help="Multiplicative factor for the RT if no RTrange present in the database. If not provided equal to 0.8."/>
+        <param label="pRTout" name="pRTout" type="float" value="0.4" help="Multiplicative factor for the RT if measured RT is outside the RTrange present in the database. If not provided equal to 0.4."/>
+        <param label="mzdCS" name="mzdCS" type="float" value="0" help="maximum mz difference allowed when computing cosine similarity scores. If one wants to use this parameter instead of ppmCS, this must be set to 0. Default 0.."/>
+        <param label="ppmCS" name="ppmCS" type="float" value="10" help="maximum ppm allowed when computing cosine similarity scores. If one wants to use this parameter instead of mzdCS, this must be set to 0. Default 10.."/>
```

Comment:
I think it will be a lot easier to have only a single ppm tolerance parameter for the user to specify and then use it consistently across the board. Like this, most people will be confused and not know where to specify what.

Reply:
Also, we can re-use some parameter definitions across tools and pull them into sections and define those in the macro file.